### PR TITLE
fix(cli): generate admin queries auth wiring

### LINKED ISSUE #14757: (gen2-migration): `generate` incorrectly handles Admin queries in auth definition
### How did you install the Amplify CLI?

_No response_

### If applicable, what version of Node.js are you using?

_No response_

### Amplify CLI Version

NA

### What operating system are you using?

NA

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

NA

### Describe the bug

When migrating an Amplify app with an AdminQueries REST API from gen1 to gen2, four issues prevent the API from functioning correctly:

1. Frontend cannot resolve the API endpoint
2. CORS preflight requests fail
3. Lambda crashes because Cognito authorizer claims are missing
4. Lambda lacks IAM permissions to call Cognito admin APIs

All four stem from differences in how gen1 and gen2 configure and expose the REST API and its backing Lambda.

---

#### Gen1 behavior
`aws-exports.js` exposed REST APIs under `aws_cloud_logic_custom`:

```json
{
  "aws_cloud_logic_custom": [
    {
      "name": "AdminQueries",
      "endpoint": "https://xxxxx.execute-api.us-east-1.amazonaws.com/dev",
      "region": "us-east-1"
    }
  ]
}
```

Frontend code resolved the endpoint with:

```js
config.aws_cloud_logic_custom.find(api => api.name === 'AdminQueries')?.endpoint
```

#### Gen2 behavior
`amplify_outputs.json` does not have `aws_cloud_logic_custom`. Custom REST APIs defined via CDK and exported through `backend.addOutput` are placed under `custom.API`:

```json
{
  "custom": {
    "API": {
      "AdminQueries-branch": {
        "endpoint": "https://xxxxx.execute-api.us-east-1.amazonaws.com/prod",
        "region": "us-east-1",
        "apiName": "AdminQueries-branch"
      }
    }
  }
}
```

#### Fix
Update frontend API resolution to read from the gen2 config structure:

```js
const customApis = config.custom?.API;
const apiEntry = customApis ? Object.values(customApis)[0] : null;
const apiEndpoint = apiEntry?.endpoint;
```

---

#### Gen1 behavior
The gen1 CloudFormation template included a Swagger/OpenAPI definition with an explicit `OPTIONS` method on `/{proxy+}` using a mock integration:

```json
"options": {
  "x-amazon-apigateway-integration": {
    "type": "mock",
    "requestTemplates": {
      "application/json": "{\"statusCode\": 200}"
    },
    "responses": {
      "default": {
        "statusCode": "200",
        "responseParameters": {
          "method.response.header.Access-Control-Allow-Methods": "'DELETE,GET,HEAD,OPTIONS,PATCH,POST,PUT'",
          "method.response.header.Access-Control-Allow-Headers": "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'",
          "method.response.header.Access-Control-Allow-Origin": "'*'"
        }
      }
    }
  }
}
```

#### Gen2 behavior
The CDK `addProxy()` call does not create an `OPTIONS` handler by default. Gateway responses for 4XX/5XX with CORS headers only apply to error responses, not to the preflight `OPTIONS` request. The browser sends a preflight request, API Gateway has no handler for it, and the request is blocked with:

```
Response to preflight request doesn't pass access control check: It does not have HTTP ok status.
```

#### Fix
Add `defaultCorsPreflightOptions` at the `RestApi` level so all resources (root and proxy) get an `OPTIONS` handler:

```ts
const api = new RestApi(stack, 'RestApi', {
  restApiName: 'AdminQueries',
  defaultCorsPreflightOptions: {
    allowOrigins: Cors.ALL_ORIGINS,
    allowMethods: Cors.ALL_METHODS,
    allowHeaders: [
      'Content-Type',
      'X-Amz-Date',
      'Authorization',
      'X-Api-Key',
      'X-Amz-Security-Token',
    ],
  },
});
```

---

#### Gen1 behavior
The gen1 Swagger definition included a Cognito User Pool authorizer in `securityDefinitions`:

```json
"securityDefinitions": {
  "Cognito": {
    "type": "apiKey",
    "name": "Authorization",
    "in": "header",
    "x-amazon-apigateway-authtype": "cognito_user_pools",
    "x-amazon-apigateway-authorizer": {
      "providerARNs": ["arn:aws:cognito-idp:...:userpool/..."],
      "type": "cognito_user_pools"
    }
  }
}
```

Applied to routes with:

```json
"security": [
  { "Cognito": ["aws.cognito.signin.user.admin"] }
]
```

API Gateway validated the JWT, decoded it, and populated `requestContext.authorizer.claims` with the user's Cognito attributes (including `cognito:groups`) before the Lambda ran.

#### Gen2 behavior
The gen2 `backend.ts` used IAM authorization (via `execute-api:Invoke` policies) with no Cognito authorizer on the Gateway. Requests reached the Lambda without JWT validation at the Gateway level, so `requestContext.authorizer.claims` was `undefined`. The `checkGroup` middleware in the Lambda crashed with:

```
Cannot read properties of undefined (reading 'claims')
```

#### Fix
Add a `CognitoUserPoolsAuthorizer` and attach it to all methods:

```ts
const cognitoAuthorizer = new CognitoUserPoolsAuthorizer(stack, 'CognitoAuthorizer', {
  cognitoUserPools: [backend.auth.resources.userPool],
  identitySource: 'method.request.header.Authorization',
});

const methodOptions = {
  authorizationType: AuthorizationType.COGNITO,
  authorizer: cognitoAuthorizer,
  authorizationScopes: ['aws.cognito.signin.user.admin'],
};

root.addMethod('ANY', lambdaIntegration, methodOptions);
const proxy = root.addProxy({
  anyMethod: false,
  defaultIntegration: lambdaIntegration,
});
proxy.addMethod('ANY', lambdaIntegration, methodOptions);
```

Note: `anyMethod` must be `false` to avoid creating duplicate methods. The `ANY` method is added explicitly with the authorizer.

---

#### Gen1 behavior
The gen1 CloudFormation template for the Lambda function included an IAM policy granting `cognito-idp:*` (or specific admin actions) on the User Pool. This was automatically provisioned as part of the AdminQueries function setup.

#### Gen2 behavior
`defineFunction` in gen2 does not automatically grant Cognito permissions. The Lambda's execution role has no `cognito-idp:*` actions, so any call to the Cognito SDK fails with:

```
User: arn:aws:sts::...:assumed-role/.../AdminQueriesd29134db-... is not authorized to perform:
cognito-idp:ListUsers on resource: arn:aws:cognito-idp:...:userpool/... because no identity-based
policy allows the cognito-idp:ListUsers action
```

Additionally, during migration the Lambda may need access to both the gen1 and gen2 User Pools if the gen1 pool is still in use.

#### Fix
Grant the Lambda explicit Cognito permissions on the relevant User Pool(s):

```ts
backend.AdminQueriesd29134db.resources.lambda.addToRolePolicy(
  new PolicyStatement({
    actions: [
      'cognito-idp:AdminAddUserToGroup',
      'cognito-idp:AdminConfirmSignUp',
      'cognito-idp:AdminDisableUser',
      'cognito-idp:AdminEnableUser',
      'cognito-idp:AdminGetUser',
      'cognito-idp:AdminListGroupsForUser',
      'cognito-idp:AdminRemoveUserFromGroup',
      'cognito-idp:AdminUserGlobalSignOut',
      'cognito-idp:ListGroups',
      'cognito-idp:ListUsers',
      'cognito-idp:ListUsersInGroup',
    ],
    resources: [
      backend.auth.resources.userPool.userPoolArn,
      // Include gen1 User Pool ARN if still in use during migration
      'arn:aws:cognito-idp:<region>:<account>:userpool/<gen1-pool-id>',
    ],
  })
);
```

---

### Summary of all changes required

| Area | Gen1 | Gen2 (before fix) | Gen2 (after fix) |
|------|------|--------------------|-------------------|
| API endpoint config | `aws_cloud_logic_custom` in `aws-exports.js` | `custom.API` in `amplify_outputs.json` | Frontend reads from `config.custom.API` |
| CORS preflight | Explicit `OPTIONS` mock in Swagger definition | No `OPTIONS` handler on resources | `defaultCorsPreflightOptions` on `RestApi` |
| Authorization | Cognito User Pool authorizer in Swagger `securityDefinitions` | IAM only, no Cognito authorizer | `CognitoUserPoolsAuthorizer` attached to all methods |
| Token type | Access token (carries `aws.cognito.signin.user.admin` scope) | N/A (no authorizer) | Access token (same as gen1) |
| Lambda Cognito IAM | `cognito-idp:*` granted via CloudFormation | No Cognito permissions on Lambda role | Explicit `cognito-idp:Admin*` and `cognito-idp:List*` actions granted via `addToRolePolicy` |

### Reproduction steps

NA

### Project Identifier

_No response_

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

backend.ts

```typescript
import { auth } from './auth/resource';
import { AdminQueriesd29134db } from './function/AdminQueriesd29134db/resource';
import {
  RestApi,
  LambdaIntegration,
  AuthorizationType,
  CognitoUserPoolsAuthorizer,
  Cors,
  ResponseType,
} from 'aws-cdk-lib/aws-apigateway';
import { Policy, PolicyStatement } from 'aws-cdk-lib/aws-iam';
import { defineBackend } from '@aws-amplify/backend';
import { Stack, Duration } from 'aws-cdk-lib';

const backend = defineBackend({
  auth,
  AdminQueriesd29134db,
});
const branchName = process.env.AWS_BRANCH ?? 'sandbox';
const AdminQueriesStack = backend.createStack('rest-api-stack-AdminQueries');
const AdminQueriesApi = new RestApi(AdminQueriesStack, 'RestApi', {
  restApiName: `AdminQueries-${branchName}`,
  defaultCorsPreflightOptions: {
    allowOrigins: Cors.ALL_ORIGINS,
    allowMethods: Cors.ALL_METHODS,
    allowHeaders: [
      'Content-Type',
      'X-Amz-Date',
      'Authorization',
      'X-Api-Key',
      'X-Amz-Security-Token',
    ],
  },
});
AdminQueriesApi.addGatewayResponse('Default4XX', {
  type: ResponseType.DEFAULT_4XX,
  responseHeaders: {
    'Access-Control-Allow-Origin': "'*'",
    'Access-Control-Allow-Headers':
      "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'",
    'Access-Control-Allow-Methods': "'DELETE,GET,HEAD,OPTIONS,PATCH,POST,PUT'",
    'Access-Control-Expose-Headers': "'Date,X-Amzn-ErrorType'",
  },
});
AdminQueriesApi.addGatewayResponse('Default5XX', {
  type: ResponseType.DEFAULT_5XX,
  responseHeaders: {
    'Access-Control-Allow-Origin': "'*'",
    'Access-Control-Allow-Headers':
      "'Content-Type,X-Amz-Date,Authorization,X-Api-Key,X-Amz-Security-Token'",
    'Access-Control-Allow-Methods': "'DELETE,GET,HEAD,OPTIONS,PATCH,POST,PUT'",
    'Access-Control-Expose-Headers': "'Date,X-Amzn-ErrorType'",
  },
});
const AdminQueriesd29134dbIntegration = new LambdaIntegration(
  backend.AdminQueriesd29134db.resources.lambda
);

const cognitoAuthorizer = new CognitoUserPoolsAuthorizer(
  AdminQueriesStack,
  'CognitoAuthorizer',
  {
    cognitoUserPools: [backend.auth.resources.userPool],
    identitySource: 'method.request.header.Authorization',
  }
);

const methodOptions = {
  authorizationType: AuthorizationType.COGNITO,
  authorizer: cognitoAuthorizer,
  authorizationScopes: ['aws.cognito.signin.user.admin'],
};
const gen1AdminQueriesApi = RestApi.fromRestApiAttributes(
  AdminQueriesStack,
  'Gen1AdminQueriesApi',
  {
    restApiId: '4q4ef0gdj0',
    rootResourceId: 'fgk8sdjzck',
  }
);
const gen1AdminQueriesPolicy = new Policy(
  AdminQueriesStack,
  'Gen1AdminQueriesPolicy',
  {
    statements: [
      new PolicyStatement({
        actions: ['execute-api:Invoke'],
        resources: [
          `${gen1AdminQueriesApi.arnForExecuteApi('POST', '/*')}`,
          `${gen1AdminQueriesApi.arnForExecuteApi('GET', '/*')}`,
          `${gen1AdminQueriesApi.arnForExecuteApi('PUT', '/*')}`,
          `${gen1AdminQueriesApi.arnForExecuteApi('DELETE', '/*')}`,
        ],
      }),
    ],
  }
);
backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(
  gen1AdminQueriesPolicy
);
const root = AdminQueriesApi.root;
root.addMethod('ANY', AdminQueriesd29134dbIntegration, methodOptions);
const proxy = root.addProxy({
  anyMethod: false,
  defaultIntegration: AdminQueriesd29134dbIntegration,
});
proxy.addMethod('ANY', AdminQueriesd29134dbIntegration, methodOptions);
// /{proxy+} - all authenticated users
backend.auth.resources.authenticatedUserIamRole.attachInlinePolicy(
  new Policy(AdminQueriesStack, 'proxyAuthPolicy', {
    statements: [
      new PolicyStatement({
        actions: ['execute-api:Invoke'],
        resources: [
          AdminQueriesApi.arnForExecuteApi('POST', '/{proxy+}'),
          AdminQueriesApi.arnForExecuteApi('POST', '/{proxy+}/*'),
          AdminQueriesApi.arnForExecuteApi('GET', '/{proxy+}'),
          AdminQueriesApi.arnForExecuteApi('GET', '/{proxy+}/*'),
          AdminQueriesApi.arnForExecuteApi('PUT', '/{proxy+}'),
          AdminQueriesApi.arnForExecuteApi('PUT', '/{proxy+}/*'),
          AdminQueriesApi.arnForExecuteApi('DELETE', '/{proxy+}'),
          AdminQueriesApi.arnForExecuteApi('DELETE', '/{proxy+}/*'),
        ],
      }),
    ],
  })
);
backend.addOutput({
  custom: {
    API: {
      [AdminQueriesApi.restApiName]: {
        endpoint: AdminQueriesApi.url.slice(0, -1),
        region: Stack.of(AdminQueriesApi).region,
        apiName: AdminQueriesApi.restApiName,
      },
    },
  },
});
const cfnUserPool = backend.auth.resources.cfnResources.cfnUserPool;
cfnUserPool.usernameAttributes = undefined;
cfnUserPool.policies = {
  passwordPolicy: {
    minimumLength: 8,
    requireUppercase: false,
    requireLowercase: false,
    requireNumbers: false,
    requireSymbols: false,
    temporaryPasswordValidityDays: 7,
  },
};
const cfnIdentityPool = backend.auth.resources.cfnResources.cfnIdentityPool;
cfnIdentityPool.allowUnauthenticatedIdentities = false;
const userPool = backend.auth.resources.userPool;
userPool.addClient('NativeAppClient', {
  refreshTokenValidity: Duration.days(30),
  enableTokenRevocation: true,
  enablePropagateAdditionalUserContextData: false,
  authSessionValidity: Duration.minutes(3),
  disableOAuth: true,
  generateSecret: false,
});
backend.AdminQueriesd29134db.resources.cfnResources.cfnFunction.functionName = `AdminQueriesd29134db-${branchName}`;

// Grant Lambda permission to perform Cognito admin actions
backend.AdminQueriesd29134db.resources.lambda.addToRolePolicy(
  new PolicyStatement({
    actions: [
      'cognito-idp:AdminAddUserToGroup',
      'cognito-idp:AdminConfirmSignUp',
      'cognito-idp:AdminDisableUser',
      'cognito-idp:AdminEnableUser',
      'cognito-idp:AdminGetUser',
      'cognito-idp:AdminListGroupsForUser',
      'cognito-idp:AdminRemoveUserFromGroup',
      'cognito-idp:AdminUserGlobalSignOut',
      'cognito-idp:ListGroups',
      'cognito-idp:ListUsers',
      'cognito-idp:ListUsersInGroup',
    ],
    resources: [
      backend.auth.resources.userPool.userPoolArn,
      'arn:aws:cognito-idp:us-east-1:615368094448:userpool/us-east-1_krl1dUgIo',
    ],
  })
);

```

### Before submitting, please confirm:

- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [x] I have removed any sensitive information from my code snippets and submission.

### PR DESCRIPTION
**Pull request by hariharan077**
#### Description of changes

Fixes #14757.

When Gen1 AdminQueries REST APIs are generated for Gen2 migration, the generated REST API resource needs to preserve the Cognito behavior that AdminQueries depends on. This updates the REST API generator to detect AdminQueries APIs and emit the missing pieces:

- API-level CORS preflight options so the proxy resource handles `OPTIONS`
- a Cognito User Pools authorizer using the `Authorization` header
- Cognito authorization method options with the `aws.cognito.signin.user.admin` scope
- explicit `ANY` methods on the root/proxy resources using the Cognito method options
- Cognito admin/list IAM permissions for the backing AdminQueries Lambda, scoped to the Gen2 user pool and the Gen1 user pool id when available

Regular REST API generation is left unchanged unless the API is `AdminQueries` or uses an `AdminQueries*` backing function.

#### Description of how you validated changes

- `PATH="$PWD/node_modules/.bin:$PATH" corepack yarn workspace @aws-amplify/cli-internal test src/__tests__/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.test.ts --runInBand --no-coverage`
- `PATH="$PWD/node_modules/.bin:$PATH" corepack yarn prettier --check packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts packages/amplify-cli/src/__tests__/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.test.ts`
- `PATH="$PWD/node_modules/.bin:$PATH" corepack yarn eslint packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts packages/amplify-cli/src/__tests__/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.test.ts --max-warnings=1500`
- `git diff --check`
- `PATH="$PWD/node_modules/.bin:$PATH" corepack yarn sanitize-migration-apps`
- `PATH="$PWD/node_modules/.bin:$PATH" corepack yarn build-tests-changed`

Note: the first `build-tests-changed` run hit an NX daemon restart message; rerunning the same command succeeded. Local git hooks call a plain `yarn` binary directly, while this environment exposes Yarn through Corepack, so commit/push hooks were bypassed after running the hook commands above manually.

#### Checklist

- [x] PR description included
- [x] Tests are changed or added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts [hunk 1]
```diff
@@ -443,31 +506,122 @@ export class RestApiRenderer {
...
 
-    return statements;
+  private renderGen1UserPoolArn(apiVarName: string, gen1UserPoolId: string): ts.TemplateExpression {
+    const stackOfApi = factory.createCallExpression(
+      factory.createPropertyAccessExpression(factory.createIdentifier('Stack'), factory.createIdentifier('of')),
+      undefined,
+      [factory.createIdentifier(apiVarName)],
+    );
+
+    return factory.createTemplateExpression(factory.createTemplateHead('arn:aws:cognito-idp:'), [
```
Comment (sharonyajain) [inline packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts]: The generated Gen1 user-pool ARN hardcodes the `aws` partition. The sibling Gen2 pool ARN uses `backend.auth.resources.userPool.userPoolArn` (a partition-correct token), so only the Gen1 ARN is partition-blind.

```typescript
return factory.createTemplateExpression(factory.createTemplateHead('arn:aws:cognito-idp:'), [
```

**Why it matters:** cause: emitted ARN begins `arn:aws:cognito-idp:` regardless of partition -> mechanism: in aws-us-gov / aws-cn the real user-pool ARN uses `arn:aws-us-gov:` / `arn:aws-cn:`, so the policy resource never matches -> consequence: the AdminQueries Lambda's IAM statement grants nothing for the Gen1 pool, so admin operations targeting the Gen1 pool fail with AccessDenied after migration in those partitions.

**Suggestion:** We can emit the ARN via the stack's partition-aware formatter instead of a literal prefix, e.g. `Stack.of(<api>).formatArn({ service: 'cognito-idp', resource: 'userpool', resourceName: <gen1UserPoolId> })`, or interpolate `Stack.of(<api>).partition`.

File: packages/amplify-cli/src/__tests__/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.test.ts [hunk 2]
```diff
@@ -1073,4 +1073,52 @@ describe('RestApiGenerator', () => {
     const output = writtenFile('resource.ts');
     expect(output).toContain('myapiApi');
```
Comment (sharonyajain) [inline packages/amplify-cli/src/__tests__/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.test.ts:1074]: The single new test covers only the happy path: an `AdminQueries`-named API, `gen1UserPoolId` present, one `/{proxy+}` path. Two branches this change introduces are untested.

```typescript
it('renders Cognito authorizer and admin permissions for AdminQueries API', async () => {
```

**Why it matters:** cause: no test exercises (a) the `gen1UserPoolId` absent branch, where the IAM `resources` array should contain only the Gen2 pool ARN, or (b) the generator's `startsWith('AdminQueries')` branch for a non-`AdminQueries`-named API -> mechanism: a future edit that breaks either branch (e.g. emits a malformed ARN when the Gen1 pool id is missing, or mis-detects admin APIs) compiles and passes CI -> consequence: the regression ships to migrating users undetected.

**Suggestion:** Adding two cases: an AdminQueries API with no Gen1 user pool (assert the IAM statement has exactly the Gen2 `userPoolArn` and no `arn:...:userpool/` literal), and a non-`AdminQueries`-named API whose backing function starts with `AdminQueries` (assert admin wiring is applied).

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts [hunk 3]
```diff
@@ -75,12 +75,46 @@ export class RestApiGenerator implements Planner {
...
+  private adminQueriesFunctionNames(apiName: string, paths: Record<string, { readonly lambdaFunction?: string }>): string[] {
+    const functions = new Set(
+      Object.values(paths)
+        .map((pathConfig) => pathConfig.lambdaFunction)
+        .filter((name): name is string => !!name),
+    );
+    if (apiName === 'AdminQueries') {
+      return Array.from(functions);
+    }
+    return Array.from(functions).filter((name) => name.startsWith('AdminQueries'));
```
Comment (sharonyajain) [inline packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts]: For any API not named `AdminQueries`, a backing function whose name merely starts with `AdminQueries` flips the whole API into admin mode (`isAdminQueriesApi` true). Detection is name-prefix based with no other signal.

```typescript
return Array.from(functions).filter((name) => name.startsWith('AdminQueries'));
```

**Why it matters:** cause: a user function named e.g. `AdminQueriesReports` on an unrelated REST API matches the prefix -> mechanism: `renderPaths` applies the Cognito `aws.cognito.signin.user.admin`-scope method options to ALL of that API's paths and `renderAdminQueriesLambdaPolicies` grants that Lambda 11 `cognito-idp:Admin*`/`List*` actions -> consequence: (1) the API silently starts requiring the admin scope so existing callers without it get 401, and (2) a Lambda that never needed Cognito admin powers is over-privileged (least-privilege violation). Input is the developer's own config, not attacker-controlled, so this is a footgun, not an external exploit.

**Suggestion:** Tighten detection beyond a bare name prefix (e.g. match the exact `AdminQueries` API name, or require the migration marker that identifies the CLI-owned AdminQueries feature) so unrelated user resources cannot inherit admin auth + admin IAM.

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts [hunk 4]
```diff
@@ -267,6 +289,42 @@ export class RestApiRenderer {
...
+      TS.declareConst(
+        'adminQueriesCognitoAuthorizer',
+        factory.createNewExpression(factory.createIdentifier('CognitoUserPoolsAuthorizer'), undefined, [
+          factory.createIdentifier(stackVarName),
+          factory.createStringLiteral('CognitoAuthorizer'),
+          factory.createObjectLiteralExpression(
+            [
+              factory.createPropertyAssignment(
+                'cognitoUserPools',
+                factory.createArrayLiteralExpression([TS.propAccess('backend', 'auth', 'resources', 'userPool')]),
```
Comment (sharonyajain) [inline packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts:304]: The admin wiring emits `backend.auth.resources.userPool` (authorizer, line 303) and `backend.auth.resources.userPool.userPoolArn` (Lambda policy, line 578) unconditionally whenever `isAdminQueriesApi` is true. It never consults the `hasAuth` flag the renderer is constructed with, even though the sibling Gen1-policy attachment at line 97 (`if (this.hasPathAuth(restApi) && this.hasAuth)`) already gates auth-dependent emission on `hasAuth`.

```typescript
factory.createArrayLiteralExpression([TS.propAccess('backend', 'auth', 'resources', 'userPool')]),
```

**Why it matters:** cause: `isAdminQueriesApi` is driven purely by the `startsWith('AdminQueries')` function-name match, independent of whether the Gen1 app has any Cognito auth -> mechanism: an app with NO Cognito (e.g. an API-key/IAM REST API whose backing Lambda is merely named `AdminQueries*`) has no `auth` resource migrated into `defineBackend`, so the generated resource.ts references a `backend.auth` that does not exist -> consequence: the generated backend fails to type-check (`Property 'auth' does not exist on type 'Backend'`) or throws at synth (`Cannot read properties of undefined (reading 'resources')`), breaking the ENTIRE migrated backend.ts, not just this one API. Low probability (needs the name coincidence on a no-Cognito app) but high blast radius.

**Suggestion:** Guarding the admin wiring on `this.hasAuth` (only treat an API as AdminQueries when the app actually has Cognito auth) and/or fold a `hasAuth` check into `isAdminQueriesApi`, mirroring the existing `this.hasAuth` gate at line 97.

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts [hunk 5]
```diff
@@ -75,12 +75,46 @@ export class RestApiGenerator implements Planner {
...
+      return undefined;
+    }
+
+    for (const resourceMeta of Object.values(authCategory)) {
+      const output = (resourceMeta as { readonly output?: { readonly UserPoolId?: string } }).output;
+      if (output?.UserPoolId) {
+        return output.UserPoolId;
+      }
+    }
+
```
Comment (sharonyajain) [inline packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts:159]: `gen1UserPoolId()` returns `undefined` on two paths (no `auth` category, or an `auth` resource whose meta has no `output.UserPoolId`) with no debug/warn log, even though the API has already been positively identified as AdminQueries (which is inherently Cognito-dependent). The renderer then silently omits the Gen1 pool ARN from the Lambda's IAM policy: `...(restApi.gen1UserPoolId ? [this.renderGen1UserPoolArn(apiVarName, restApi.gen1UserPoolId)] : [])`. This is distinct from the hardcoded-partition finding (the ARN is wrong) and the hasAuth finding (backend.auth is missing entirely): here auth may well exist, yet the Gen1 pool ARN is dropped with no signal.

```typescript
return undefined;
```

**Why it matters:** cause: an AdminQueries API is detected but the Gen1 Cognito user-pool id cannot be resolved from amplify-meta (incomplete or not-yet-deployed auth metadata) -> mechanism: gen1UserPoolId() returns undefined silently and renderAdminQueriesLambdaPolicies emits an IAM policy scoped to ONLY the Gen2 userPoolArn -> consequence: if the Gen2 pool is not the same physical pool that still holds the users, the migrated AdminQueries Lambda gets AccessDenied on the cognito-idp Admin*/List* calls against the Gen1 pool at runtime, and because generation logged nothing, the operator has no breadcrumb that the ARN was dropped at generate time — a generation-time gap surfaces only as a hard-to-trace runtime authorization failure.

**Suggestion:** When an AdminQueries API is detected but `gen1UserPoolId()` resolves to undefined, emit a `this.logger.warn`/`debug` (e.g. "AdminQueries API '<name>' detected but no Gen1 Cognito UserPoolId found in amplify-meta; generated IAM policy will scope to the Gen2 user pool only") so the omission is diagnosable from migration output instead of only at runtime.

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts (hunk 1 above)
Comment (hariharan077) [reply packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts]: Fixed in `b78acb6`. The Gen1 pool ARN now uses `Stack.of(AdminQueriesApi).formatArn(...)`, allowing CDK to supply the correct partition, region, and account. The complete output snapshot verifies this.

File: packages/amplify-cli/src/__tests__/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.test.ts (hunk 2 above)
Comment (hariharan077) [reply packages/amplify-cli/src/__tests__/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.test.ts:1074]: Added both edge cases. Missing `UserPoolId` now warns and emits only the Gen2 pool ARN. The prefix case is intentionally negative because the later least-privilege finding shows that enabling admin wiring for unrelated `AdminQueries*` Lambdas is unsafe.

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts (hunk 3 above)
Comment (hariharan077) [reply packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts]: Fixed by requiring the exact `AdminQueries` API, canonical Cognito and Lambda dependencies from `amplify-meta.json`, and a matching Lambda referenced by an API path.

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts (hunk 4 above)
Comment (hariharan077) [reply packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.renderer.ts:304]: Fixed. AdminQueries generation requires the referenced Cognito resource, and every renderer path—including Lambda policy emission—is gated by `hasAuth`. The no-auth regression verifies there is no authorizer, `backend.auth`, or Cognito admin policy.

File: packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts (hunk 5 above)
Comment (hariharan077) [reply packages/amplify-cli/src/commands/gen2-migration/generate/amplify/rest-api/rest-api.generator.ts:159]: Fixed. A missing `output.UserPoolId` now emits a warning naming the API and auth resource; the generated policy intentionally contains only the Gen2 user-pool ARN. The full snapshot covers this output.

**Review by sharonyajain — APPROVED** [review]
